flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -148,9 +147,6 @@ else if (pKeys instanceof Keys.SelectorFunctionKeys) {
 			}
 			
 		} 
-		else if (pMethod == PartitionMethod.RANGE) {
```

Comment:
We need to add a check here to ensure that `pKeys.getKeyType().isSortKeyType() == true`. The reason for this is that some keys can be used for grouping but not for sorting because their order is not well defined. For example POJO types have multiple fields on which they can be grouped, but the order of their fields is not defined so the order of POJO objects isn't defined either.